core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanOps.scala

```diff
@@ -168,4 +185,28 @@ object SpanOps {
     def apply[F[_]](span: Span[F], trace: F ~> F): Res[F] =
       Impl(span, trace)
   }
+
+  /** Implementation for [[SpanOps.mapK]]. */
+  private class MappedK[F[_]: MonadCancelThrow, G[_]: MonadCancelThrow](
+      ops: SpanOps[F]
+  )(implicit kt: KindTransformer[F, G])
+      extends SpanOps[G] {
+    def startUnmanaged: G[Span[G]] =
+      kt.liftK(ops.startUnmanaged).map(_.mapK[G])
+
+    def resource: Resource[G, Res[G]] =
+      Resource {
+        kt.liftK {
+          ops.resource.allocated
+            .map { case (res, release) =>
+              res.mapK[G] -> kt.liftK(release)
+            }
+        }
+      }
```

Comment:
Seems to be identical to (unless I'm missing something):
```suggestion
      ops.resource.mapK(kt.liftK).map(res => res.mapK[G])
```

Reply:
... I do believe you're correct. I looked at natchez for inspiration; I'm not sure why it uses `allocated` and manually constructs a new `Resource` 🤷‍♀️